File: 0509-fibonacci-number/0509-fibonacci-number.js
/**
 * @param {number} n
 * @return {number}
 */
var fib = function(n) {
    if (n < 2) return n;
    const mem = [0, 1];
    while (n > 2) {
        mem.push(mem[0] + mem[1]);
        mem.shift();
        n--;
    }
    return mem[0] + mem[1];
};

File: 0509-fibonacci-number/0509-fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        if n < 2:
            return n
        mem = [0, 1]
        while n > 2:
            mem.append(mem[0]+mem[1])
            del mem[0]
            n -= 1
        return mem[0]+mem[1]